satpy/enhancements/__init__.py

```
@@ -231,33 +227,60 @@ def create_colormap(palette):
-                                   data[i, 2] / 255.)))
-        return Colormap(*cmap)
-
-    colors = palette.get('colors', None)
-    if isinstance(colors, (tuple, list)):
+        cols = data.shape[1]
+        default_modes = {
+            3: 'RGB',
+            4: 'VRGB',
+            5: 'VRGBA'
+        }
+        default_mode = default_modes.get(cols)
+        mode = palette.setdefault('colormap_mode', default_mode)
+        if mode is None or len(mode) != cols:
+            raise ValueError(
+                "Unexpected colormap shape for mode '{}'".format(mode))
+
+        rows = data.shape[0]
+        if mode[0] == 'V':
+            colors = data[:, 1:]
+            if color_scale != 1:
+                colors = data[:, 1:] / float(color_scale)
+            values = data[:, 0]
+        else:
+            colors = data
+            if color_scale != 1:
+                colors = colors / float(color_scale)
+            values = np.arange(rows) / float(rows - 1)
+        cmap = Colormap(*zip(values, colors))
+    elif isinstance(colors, (tuple, list)):
         cmap = []
         values = palette.get('values', None)
         for idx, color in enumerate(colors):
             if values is not None:
                 value = values[idx]
             else:
                 value = idx / float(len(colors) - 1)
+            if color_scale != 1:
+                color = tuple(elem / float(color_scale) for elem in color)
             cmap.append((value, tuple(color)))
-        return Colormap(*cmap)
-
-    if isinstance(colors, str):
+        cmap = Colormap(*cmap)
+    elif isinstance(colors, str):
         from trollimage import colormap
         import copy
-        return copy.copy(getattr(colormap, colors))
+        cmap = copy.copy(getattr(colormap, colors))
+    else:
+        raise ValueError("Unknown colormap format: {}".format(palette))
+
+    if palette.get("reverse", False):
+        cmap.reverse()
+    if 'min_value' in palette:
```

Should we check if `max_value` is present also ?
```suggestion
    if 'min_value' in palette and 'max_value' in palette:
```

---

So the only issue with that is if they specify one of them and not both then it should probably raise an error message, right?

---

In principle yes.